src/lib/y2partitioner/device_graphs.rb

```diff
@@ -0,0 +1,47 @@
+require "singleton"
+
+module Y2Partitioner
+  # Object that holds and manipulates of storage device graphs.
+  class DeviceGraphs
```

Comment:
@mvidner @jreidinger and that is exactly what we are currently doing, the main problem was that we was working directly with @staging instead of a @staging copy, so I just to initialize current with a copy of staging, and from that point until the end of partitioner when we will write it to `staging` in case of write.